src/arvenums.h

```diff
@@ -81,6 +81,12 @@ typedef guint32 ArvPixelFormat;
 
 #define ARV_PIXEL_FORMAT_BIT_PER_PIXEL(pixel_format) (((pixel_format) >> 16) & 0xff)
 
+#define ARV_PIXEL_FORMAT_8_BIT(pixel_format) ((pixel_format & 0x00080000) && !(pixel_format & ~0x0318003f))
```

Comment:
What is wrong with ARV_PIXEL_FORMAT_BIT_PER_PIXEL ? What is the need for your new 3 macros ?

Reply:
ARV_PIXEL_FORMAT_BIT_PER_PIXEL gives the number of bits used during transport, e.g.:

Mono8: 8
Mono10: 16
Mono12: 16

I'd like to know the bit depth of the camera's ADC, e.g.:

Mono8: 8
Mono10: 10
Mono12: 12

I tried (and failed) to make simple macros which tell me if a ArvPixelFormat corresponds to an 8bit/10bit/12bit ADC.

Reply:
Ok.

Did you read https://www.emva.org/wp-content/uploads/GenICam_PFNC_1_1_01.pdf ?

It looks like it is not possible to extract the bit depth from the pixel format 32bit identifier. I fear it has to be handled with a dedicated table, which could also include packing and location informations, based on the PFNC document.